modules/messages/server/controllers/messages.server.controller.js

```diff
@@ -63,57 +63,88 @@ function sanitizeMessages(messages) {
 }
 
 /**
- * Sanitize threads
- * @param {Array} threads - list of threads to go trough
- * @param {ObjectId} authenticatedUserId - ID of currently authenticated user
- * @returns {Array}
+ * Fill `userFrom` or `userTo` with deleted user's ID
+ *
+ * If user is deleted, the object from `User` collection is gone.
+ * `Thread.paginate()` attempts to populate it but failing to do so,
+ * leaves `userFrom` or `userTo` as `null` instead of just keeping
+ * the original document `_id` which would still be useful at the frontend.
+ * Until we refactor our code not to use `mongoose-paginate`, this helper
+ * will backfill the IDs. Cumbersome and not optimal, but there are not that
+ * many deleted profiles that this would be a significant issue.
+ *
+ * @param {Object} Thread
+ * @return {Object} Thread with possibly mutated `thread.userFrom` / `thread.userTo`
  */
-function sanitizeThreads(threads, authenticatedUserId) {
-
+function fillDeletedProfiles(threads, callback) {
   if (!threads || !threads.length) {
-    return [];
+    return callback(null, []);
   }
 
-  // Sanitize each outgoing thread
-  var threadsCleaned = [];
+  async.mapSeries(
+    threads,
+    function (thread, done) {
+      if (thread.userTo && thread.userFrom) {
+        return done(null, thread);
+      }
 
-  threads.forEach(function (thread) {
+      // Check which field is `null` — only one of them can
+      var field = !thread.userTo ? 'userTo' : 'userFrom';
 
-    // Threads need just excerpt
-    thread = thread.toObject();
+      Thread.findById(thread._id, field, function (err, newThread) {
+        if (!err && newThread) {
+          thread[field] = {
+            _id: newThread[field]
+          };
+        }
+        done(null, thread);
+      });
+    },
+    callback
+  );
+}
 
-    // Clean message content from html + clean all whitespace + shorten
-    if (thread.message) {
-      thread.message.excerpt = thread.message.content ? textServiceService.plainText(thread.message.content, true).substring(0, 100).trim() + ' …' : '…';
-      delete thread.message.content;
-    } else {
-      // Ensure this works even if messages couldn't be found for some reason
-      thread.message = {
-        excerpt: '…'
-      };
-    }
+/**
+ * Sanitize threads
+ *
+ * @param {Array} threads - list of threads to go trough
+ * @param {ObjectId} authenticatedUserId - ID of currently authenticated user
+ * @returns {Array}
+ */
+function sanitizeThreads(threads, authenticatedUserId, callback) {
```

Comment:
Refactoring `sanitizeThreads()` is prolly better to do outside this PR for clarity.